sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -777,6 +777,39 @@ def struct_contains_unknowns(props_struct: struct_pb2.Struct) -> bool:
     return False
 
 
+def _unwrap_rpc_secret_struct_properties(
+    value: Any, on_secret_encountered: Callable
```

Comment:
Using a callback seems a little unusual for this, but it's fine. I probably would have had the function return two values: the result and a bool value indicating whether there were any secrets.

Reply:
> I probably would have had the function return two values

✅

Reply:
Done